### PR TITLE
metrics: Add metrics and telemetry for aggressive locking

### LINKED ISSUE #41037: Add metrics and telemetry for aggressive locking
## Enhancement

Ref: https://github.com/tikv/tikv/issues/13298

This feature introduced a new locking mechanism named aggressive locking to TiDB. To know whether it works, we need to add metrics and telemetry to it.

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve?

Issue Number: close #41037
ref https://github.com/tikv/tikv/issues/13298
Requires https://github.com/tikv/client-go/pull/687

Problem Summary:

### What is changed and how it works?

This PR adds metrcis and telemetry for usages of aggressive locking.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/adapter.go [hunk 1]
```diff
@@ -86,6 +86,11 @@ var (
 	selectForUpdateRetryDuration        = metrics.PessimisticDMLDurationByAttempt.WithLabelValues("select-for-update", "retry")
 	dmlFirstAttemptDuration             = metrics.PessimisticDMLDurationByAttempt.WithLabelValues("dml", "first-attempt")
 	dmlRetryDuration                    = metrics.PessimisticDMLDurationByAttempt.WithLabelValues("dml", "retry")
+
```
Comment (cfzjywxk) [inline executor/adapter.go:89]: We may add some comments to explain about the meaning of `Used` and `Effective`.

Comment (ekexium) [reply executor/adapter.go:89]: Also in TxnCtx. These 2 names can be confusing

File: metrics/session.go [hunk 2]
```diff
@@ -178,6 +178,14 @@ var (
 			Help:      "Bucketed histogram of duration of pessimistic DMLs, distinguished by first attempt and retries",
 			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 28), // 1ms ~ 1.5days
 		}, []string{LblType, LblPhase})
+
+	AggressiveLockingUsageCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "server",
+			Name:      "transaction_aggressive_locking_usage",
+			Help:      "The counter of statements and transactions that aggressive locking is used or takes effect",
```
Comment (ekexium) [inline metrics/session.go:187]:
```suggestion
			Help:      "The counter of statements and transactions in which aggressive locking is used or takes effect",
```

**Review by ekexium — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Comment by MyonKeminta** [comment]
> tide Pending — Not mergeable. Retesting: idc-jenkins-ci-tidb/unit-test

¿

**Comment by cfzjywxk** [comment]
@wuhuizuo 
Need help it's not merged.

**Pull request merged**